### PR TITLE
Add memory stats to the X-Ray trace

### LINKED ISSUE #71: Memory profiling in X-Ray
It would be very helpful if we were able to track the total memory for the trace and memory usage and % of the total memory for individual segments to help memory profiling.

This will probably need memory usage stats to be available from Xhprof. Tideways's Xhprof fork does add this memory functionality, just for a reference - https://github.com/tideways/php-xhprof-extension/pull/52

### PR DESCRIPTION
**Pull request by Nikschavan**
Fix #71

### CODE REVIEW

**Review by joehoyle — APPROVED** [review]

File: inc/namespace.php [hunk 1]
```diff
@@ -425,7 +425,12 @@ function ( $error ) {
 		'stats'     => $stats,
 	];
 
-	$trace['metadata'] = redact_metadata( $metadata );
+	$annotations = [
+		'memoryUsage' => round( memory_get_peak_usage() / 1048576, 2 ), // Convert bytes to mb.
```
Comment (rmccue) [inline inc/namespace.php:429]: Can we send the actual value and process it on output instead?

File: inc/namespace.php [hunk 2]
```diff
@@ -425,7 +425,12 @@ function ( $error ) {
...
+	];
+
+	$trace['metadata']    = redact_metadata( $metadata );
```
Comment (rmccue) [inline inc/namespace.php:432]: This shouldn't be aligned, as alignment can cause unnecessary code churn in future changes.
```suggestion
	$trace['metadata'] = redact_metadata( $metadata );
```

**Review by rmccue — CHANGES_REQUESTED** [review]

File: inc/namespace.php (hunk 1 above)
Comment (joehoyle) [reply inc/namespace.php:429]: Counter point is that being able to include this in the filter expressions, having it atleast as a mb value would be beneficial. Maybe not the rounding though

Comment (Nikschavan) [reply inc/namespace.php:429]: I considered this but I think that caused an issue in UX when you actually try searching in X-ray using annotations, You will see values in MB in the UI but when searching you will have to convert it to KB.

Let me know what you think

Comment (rmccue) [reply inc/namespace.php:429]: I'd say we can convert the units in the filtering side if we want? e.g. `annotations.memoryUsage > 2M` or `2G` etc

Comment (joehoyle) [reply inc/namespace.php:429]: I think it's better to stay native, else we'd need to transform the filter expression further. Also I don't think we'll ever need anything other than MB units for memory usage

**Review by rmccue — APPROVED** [review]
Disagree and commit ;)

File: inc/namespace.php (hunk 1 above)
Comment (Nikschavan) [reply inc/namespace.php:429]: Should I remove rounding from the conversion?

Comment (Nikschavan) [reply inc/namespace.php]: Removed the rounding-off of the used memory to get the exact value of memory consumption

**Review by joehoyle — APPROVED** [review]

**Pull request merged**